spec/app_spec.rb

```diff
@@ -0,0 +1,123 @@
+require_relative '../app.rb'
+require_relative '../lib/item_repository.rb'
+require_relative '../lib/order_repository.rb'
+
+RSpec.describe Application do
+
+  def reset_all_tables
+    seed_sql = File.read('spec/seeds.sql')
+    connection = PG.connect({ host: '127.0.0.1', dbname: 'shop_items_orders_test' })
+    connection.exec(seed_sql)
+  end
+  
+  before(:each) do 
+    reset_all_tables
+  end
+
+  it "lists all items" do
+    
+    item_repository = ItemRepository.new
+    order_repository = double :order_repository
+    io = double :io
+    app = Application.new('shop_items_orders_test', io, item_repository, order_repository)
+
+    expect(io).to receive(:puts).with("Welcome to the shop management program!")
+    expect(io).to receive(:puts).with("\n")
+    expect(io).to receive(:puts).with("What do you want to do?")
+    expect(io).to receive(:puts).with(" 1 = list all shop items")
+    expect(io).to receive(:puts).with(" 2 = list all items in an order")
+    expect(io).to receive(:puts).with(" 3 = list all shop orders")
+    expect(io).to receive(:puts).with(" 4 = list all orders containing an item")
+    expect(io).to receive(:puts).with("\n")
+    expect(io).to receive(:print).with("Enter your choice: ")
+    expect(io).to receive(:gets).and_return("1")
+    expect(io).to receive(:puts).with("\n")
+
+    result = app.run
+
+    expect(result.first).to eq "#1 Item 1 - Unit price: 12.0000 - Quantity: 5"
+
+  end
+
+  it "lists orders related to a specific item" do
+    
+    order_repository = OrderRepository.new
```

Comment:
Line 44, 45, 73, 97, 98: As this is app_spec, create doubles for other classes instead of making a new instance of the class, so that we are only using the app class in the tests